### DISCUSSION TITLE
Rollback an update

### DISCUSSION THREAD

**Discussion by beckhamrryyaann** [IBM/ibmi-tobi#150] [Q&A] [answered]
How do i roll back a Bob update.  I guess this is a more general question as How do i rollback any open source package update?

Also I see the latest version for Bob on Github is 2.4.4 but when i go to ACS open source package manager I can only see the 2.4.1 update?  How does that work?  Should the version on Github always match what is available in the IBM i repo to pick up ?  

Thanks!
Ryan

**Comment by edmundreinhardt** [comment, marked as answer]
sorry about that.  I have a fix rolling out tomorrow

To roll back simply do
`yum downgrade bob`

**Comment by edmundreinhardt** [comment]
![image](https://user-images.githubusercontent.com/11184579/196552264-75098a1f-21ec-4246-acaa-d163d759fbd6.png)

**Comment by edmundreinhardt** [comment]
I have been actively trying to address regressions that I have found and I have put the fix into ibmi-bob repo.  There is an additional step that IBM Rochester has to do to get my published release into the yum repository.
For those who are impatient they can always download the rpm from this repo and install it via `rpm -U <filename>`

**Reply by edmundreinhardt** [reply]
This would require have rpm installed via
`yum install rpm-build`
as described on
https://ibm.github.io/ibmi-bob/#/contributing/build-rpm

**Comment by beckhamrryyaann** [comment]
Thanks Edmund.  That is good information to know.    I didn’t mean to come across as in patient.  I appreciate all the work IBM and your team is putting into Bob.  Sent from my iPhoneOn Oct 18, 2022, at 6:02 PM, Edmund Reinhardt ***@***.***> wrote:﻿
I have been actively trying to address regressions that I have found and I have put the fix into ibmi-bob repo.  There is an additional step that IBM Rochester has to do to get my published release into the yum repository.
For those who are impatient they can always download the rpm from this repo and install it via rpm -U <filename>

**Reply by edmundreinhardt** [reply]
I did not mean that in a bad way, Ryan :-)   Just giving options and education on the process.   First we create the rpm which is installable as is and we test it and then attach it to the release.  Then IBM picks it up and puts it into the yum repository.  This last step I am hoping for today.

**Reply by beckhamrryyaann** [reply]
Hey Edmund,  Thanks for the knowledge transfer.  While we are on it, if I do `yum --showduplicates list bob` it only shows releases 2.3.9 and 2.4.1 as available releases for me to install.  I would expect to see all previous releases? But I'm not real sure what I should see??   Not sure why it would skip release 2.3.10 and show 2.3.9.   Thanks again.

![image](https://user-images.githubusercontent.com/6051612/196728212-6af1b933-84cb-4e37-b960-84996c1eacd9.png)

**Reply by edmundreinhardt** [reply]
I am not a full yum expert, but I wonder if those are the versions you have installed on your IBM i

**Comment by edmundreinhardt** [comment]
FYI, `yum install bob` will get you the new 2.4.4 version with all the fixes you are looking for.
@beckhamrryyaann

**Reply by beckhamrryyaann** [reply]
I am still showing Bob 2.4.1 is the latest release?  Do i need to update my repos list or something to make sure i'm seeing the latest?  i am also using the ACS open package manager and it says the same thing.  2.4.1 is the latest.  

![image](https://user-images.githubusercontent.com/6051612/197019987-7176161f-2c2b-4486-8fe8-e6b391da4a59.png)

**Comment by edmundreinhardt** [comment]
as per https://ibm.github.io/ibmi-bob/#/getting-started/installation
try
```
yum install ibmi-repos
```
to get the  latest repo level

**Reply by beckhamrryyaann** [reply]
Its says that is current as well .... sorry to be a pest here.  This may help future developers when they start learning Bob so I'll keep asking what I hope is good questions.  :-)

![image](https://user-images.githubusercontent.com/6051612/197021469-f646ad63-640c-4a9f-9c65-3cf1f08edbc3.png)

**Comment by edmundreinhardt** [comment]
Asking experts on that.

By the way on your previous question on downgrade, the expert says
```
The way the default deploy works, it removes old versions, but if you want I can manually redeploy older versions too that way users could rollback
```

**Reply by beckhamrryyaann** [reply]
Good to know.  I'll leave that up to you guys on what is kept and what is removed.  My only opinion is only the best builds should be out there. I know all builds should be the best but reality is some are more stable than others even in the best of cases.   2.4.1 should not :-).   and I seem to remember I had to upgrade from 2.3.9 to 2.4.0 to fix several issues.

**Comment by edmundreinhardt** [comment]
What level of IBM i are you on?

**Reply by worksofliam** [reply]
Hi @beckhamrryyaann!

Let's try a couple of things

* `yum clean all` to remove the cache items. Maybe something got stuck
* `yum list available` should produce the new version

```
bob.ppc64                                 2.4.4-1           Artifactory_7.3    
```

Then you should be able to do `yum install bob`. Let us know how you get on

**Reply by beckhamrryyaann** [reply]
Great stuff guys!  Following Liam's two commands above brought in 2.4.4.  Thanks all for the help.
I am running IBM i 7.4 TR4.

**Reply by worksofliam** [reply]
@beckhamrryyaann Love it!! Enjoy.
